extensions/cornerstone-dicom-sr/src/viewports/OHIFCornerstoneSRViewport.tsx

```diff
@@ -420,18 +427,14 @@ async function _getViewportReferencedDisplaySetData(
   measurementSelected,
   displaySetService
 ) {
-  if (!displaySet.keyImageDisplaySet) {
-    // Create a new display set, and preserve a reference to it here,
-    // so that it can be re-displayed and shown inside the SR viewport.
-    // This is only for ease of redisplay - the display set is stored in the
-    // usual manner in the display set service.
-    displaySet.keyImageDisplaySet = createReferencedImageDisplaySet(
-      displaySetService,
-      displaySet
-    );
-  }
+  const { measurements } = displaySet;
+  const measurement = measurements[measurementSelected];
+
+  const { displaySetInstanceUID } = measurement;
 
-  const referencedDisplaySet = displaySet.keyImageDisplaySet;
+  const referencedDisplaySet = displaySetService.getDisplaySetByUID(
```

Comment:
I believe `createReferencedImageDisplaySet.ts` is no longer referenced/used. Consider removing it, but at your discretion.